### PR TITLE
Remove soundspeed from openboundary system

### PR DESCRIPTION
**Pull request by svchb**
Saving sound speed in the openboundary system makes no sense especially in the context of multiple fluids that can be at the openboundary.
As such remove sound speed and reuse the information from the fluid system(s).

Extracted from #604

### CODE REVIEW

**Review by efaulhaber — APPROVED** [review]
Looks reasonable to me, but I would like to let @LasNikas have a look.

File: src/schemes/boundary/open_boundary/method_of_characteristics.jl [hunk 1]
```diff
@@ -10,9 +10,11 @@ struct BoundaryModelLastiwka end
 
 @inline function update_quantities!(system, boundary_model::BoundaryModelLastiwka,
                                     v, u, v_ode, u_ode, semi, t)
-    (; density, pressure, cache, flow_direction, sound_speed,
+    (; density, pressure, cache, flow_direction,
     reference_velocity, reference_pressure, reference_density) = system
 
+    sound_speed = system_sound_speed(system)
```
Comment (LasNikas) [inline src/schemes/boundary/open_boundary/method_of_characteristics.jl:16]:
```suggestion
    sound_speed = system_sound_speed(system.fluid_system)
```

File: src/schemes/boundary/open_boundary/method_of_characteristics.jl [hunk 2]
```diff
@@ -123,6 +125,7 @@ function evaluate_characteristics!(system, neighbor_system::FluidSystem,
 
     system_coords = current_coordinates(u, system)
     neighbor_coords = current_coordinates(u_neighbor_system, neighbor_system)
+    sound_speed = system_sound_speed(system)
```
Comment (LasNikas) [inline src/schemes/boundary/open_boundary/method_of_characteristics.jl:128]:
```suggestion
    sound_speed = system_sound_speed(system.fluid_system)
```

File: src/schemes/boundary/open_boundary/system.jl [hunk 3]
```diff
@@ -367,3 +365,4 @@ end
 @inline viscosity_model(system::OpenBoundarySPHSystem, neighbor_system::BoundarySystem) = neighbor_system.boundary_model.viscosity
 # When the neighbor is an open boundary system, just use the viscosity of the fluid `system` instead
 @inline viscosity_model(system, neighbor_system::OpenBoundarySPHSystem) = system.viscosity
+@inline system_sound_speed(system::OpenBoundarySPHSystem) = system_sound_speed(system.fluid_system)
```
Comment (LasNikas) [inline src/schemes/boundary/open_boundary/system.jl:368]:
```suggestion
```

**Review by LasNikas — CHANGES_REQUESTED** [review]
Good point thanks. 
IMO, we then don't need a wrapper for the `OpenBoundarySPHSystem`

**Pull request merged**